scripts/_common.py

```diff
@@ -0,0 +1,262 @@
+    if (
+        tldr_root := next((path for path in f.parents if path.name == "tldr"), None)
+    ) is not None:
+        return tldr_root
+    elif "TLDR_ROOT" in os.environ:
+        return Path(os.environ["TLDR_ROOT"])
+    raise SystemExit(
+        "\x1b[31mPlease set TLDR_ROOT to the location of a clone of https://github.com/tldr-pages/tldr."
+    )
+
+
+def test_get_tldr_root():
+    tldr_root = get_tldr_root("/path/to/tldr/scripts/test_script.py")
+    assert tldr_root == Path("/path/to/tldr")
+
+    # Set TLDR_ROOT in the environment
+    os.environ["TLDR_ROOT"] = "/path/to/tldr_clone"
+
+    tldr_root = get_tldr_root("/tmp")
+    assert tldr_root == Path("/path/to/tldr_clone")
+
+    del os.environ["TLDR_ROOT"]
+
+    # Remove TLDR_ROOT from the environment
+    original_env = os.environ.pop("TLDR_ROOT", None)
+
+    # Check if SystemExit is raised
+    raised = False
+    try:
+        get_tldr_root("/tmp")
+    except SystemExit:
+        raised = True
+    assert raised
+
+    # Restore the original values
+    if original_env is not None:
+        os.environ["TLDR_ROOT"] = original_env
+
+
+def get_pages_dir(root: Path):
+    return [d for d in root.iterdir() if d.name.startswith("pages")]
+
+
+def test_get_pages_dir():
+    # Create temporary directories with names starting with "pages"
+
+    root = Path("test_root")
+    root.mkdir(exist_ok=True)
+
+    # Create temporary directories with names that do not start with "pages"
+    (root / "other_dir_1").mkdir()
+    (root / "other_dir_2").mkdir()
+
+    # Call the function and verify that it returns an empty list
+    result = get_pages_dir(root)
+    assert result == []
+
+    (root / "pages_1").mkdir()
+    (root / "pages_2").mkdir()
+    (root / "other_dir").mkdir()
+
+    # Call the function and verify the result
+    result = get_pages_dir(root)
+    expected = [root / "pages_1", root / "pages_2"]
+    assert result == expected
+
+    for item in root.glob("**/*"):
+        item.rmdir()
+    root.rmdir()
+
+
+def get_locale(path: Path) -> str:
+    """
+    Get the locale from the path.
+    """
+
+    # compute locale
+    pages_dirname = path.parents[1].name
+    if "." in pages_dirname:
+        _, locale = pages_dirname.split(".")
+    else:
+        locale = "en"
+
+    return locale
+
+
+def test_get_locale():
+    assert get_locale(Path("path/to/pages.fr/common/tldr.md")) == "fr"
+
+    assert get_locale(Path("path/to/pages/common/tldr.md")) == "en"
+
+    assert get_locale(Path("path/to/other/common/tldr.md")) == "en"
+
+
+def get_status(action: str, dry_run: bool, type: str) -> str:
+    """
+    Get a colored status line.
+
+    Parameters:
+    action (str): The action to perform.
+    dry_run (bool): Whether to perform a dry-run.
+    type (str): The kind of object to modify (alias, link).
+
+    Returns:
+    str: A colored line
+    """
+
+    match action:
+        case "added":
+            status_prefix = "\x1b[36m"  # Cyan color
```

Comment:
## Suggestion
How about adding a simple enum for colors in `_common.py`?

```python
from enum import Enum
```

```
class Colors(str, Enum):
    def __str__(self):
         return str(self.value) # make str(Colors.COLOR) return the ANSI code instead of an Enum object

    CYAN = "x1b[36m"
    BLUE = "x1b[34m"
    RED = ''x1b[31m"
    RESET = "x1b[0m"
```
(We can add more colors if needed)

```suggestion
            status_prefix = Colors.CYAN
```
Other parts of this script and the others would have to be refactored too.

## Why?

There's so much code that uses colors and it would not be a good idea to add a comment about the colors in all those places, so an Enum makes the color explicit without the need to always commenting the line.